tests/model_registry/test_user_permission.py

```diff
@@ -0,0 +1,292 @@
+                    "registriesNamespace": TEST_NAMESPACE,
+                },
+            }
+        })
+    ],
+    indirect=True,
+)
+class TestUserPermission:
+    """
+    Test suite for verifying user and group permissions for the Model Registry.
+    """
+
+    @pytest.mark.smoke
+    @pytest.mark.parametrize(
+        "user_name, password, context_manager",
+        [
+            ("htpasswd-cluster-admin-user", os.environ.get("ADMIN_PASSWORD"), nullcontext()),
+            ("ldap-user1", os.environ.get("NON_ADMIN_PASSWORD"), pytest.raises(ForbiddenException)),
+        ],
+    )
+    def test_user_permission(
+        self: Self,
+        updated_dsc_component_state_scope_class: Namespace,
+        model_registry_instance: ModelRegistry,
+        model_registry_namespace: str,
+        admin_client: DynamicClient,
+        user_name: str,
+        password: str,
+        context_manager: ContextManager,
+    ):
+        """
+        Test that a user with permission can access the Model Registry,
+        and a user without permission receives a ForbiddenException.
+        """
+        assert model_registry_instance.name == MR_INSTANCE_NAME
+        user_token = get_token(user_name=user_name, password=password, admin_client=admin_client)
+
+        assert_mr_client(
+            user_token=user_token,
+            admin_client=admin_client,
+            context=context_manager,
+            mr_instance=model_registry_instance,
+            mr_namespace_name=model_registry_namespace,
+        )
+
+    @pytest.mark.smoke
+    @pytest.mark.parametrize(
+        "user_name, password",
+        [
+            ("ldap-user1", os.environ.get("NON_ADMIN_PASSWORD")),
+        ],
+    )
+    def test_user_added_to_group(
+        self: Self,
+        updated_dsc_component_state_scope_class: Namespace,
+        model_registry_instance: ModelRegistry,
+        model_registry_namespace: str,
+        admin_client: DynamicClient,
+        user_name: str,
+        password: str,
+        user_in_group_context: Callable[[str], ContextManager],
+    ):
+        """
+        Test that a user cannot access the Model Registry before being added to a group,
+        and can access it after being added to the group.
+        """
+        assert model_registry_instance.name == MR_INSTANCE_NAME
+
+        user_token = get_token(user_name=user_name, password=password, admin_client=admin_client)
+
+        LOGGER.info("User has no access to MR")
+        assert_mr_client(
+            user_token=user_token,
+            admin_client=admin_client,
+            context=pytest.raises(ForbiddenException),
+            mr_instance=model_registry_instance,
+            mr_namespace_name=model_registry_namespace,
+        )
+
+        LOGGER.info("Add user to the model registry users group")
+        with user_in_group_context(user_name):
+            user_token = get_token(user_name=user_name, password=password, admin_client=admin_client)
```

Comment:
IIRC after adding a user to a group it takes a while to get "reconciled" such that the token works as expected. Is there some way to wait until the reconciliation is successful? and/or adding a retry to the assert_mr_client call